VerilogCodingStyle.md

```diff
@@ -1808,6 +1808,62 @@ reduction are significant enough to outweigh the possible synthesis mismatch
 issues that the use of `X` literals may entail (especially with the gate-counts
 available in today's technologies).
 
+
+#### Catching errors where invalid values are consumed
+
+One use case for `X` is where a signal is driven to `X` when it is invalid and
+shouldn't be consumed. For example on a bus interface the address lines may be
+driven to `X` when a valid transaction isn't present. Anything that incorrectly
+uses that invalid address will itself produce an `X` value that will be caught
+by an appropriate `` `ASSERT_KNOWN `` and trigger a simple to diagnose failure.
+
+For any value that could be invalid but won't be driven to `X` (as they are
+internal signals rather than generated by a DV environment) and is used to
+trigger some action (e.g. it factors into a enable for a register write) it is
+recommended an assertion is added to check that if the action is triggered then
+the value is valid. This triggers a similar simple to diagnose failure when an
+invalid value has been accidentally used.
```

Comment:
The first sentence is very clause-heavy. Perhaps a simpler way to write it:

For an internally-generated signal that could be invalid (but not driven to `X`)
and is used to enable another signal (such as a register write-enable), it is
recommended to add an assert to check that when the enable is true, the signal
is valid. This triggers ...